wv2 RelNotes for Release SDK 151 (Aug. 2026) (#3855)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/release-notes/index.md

   * [ICoreWebView2ExperimentalEnvironment16::CreateDiagnosticMonitor](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalenvironment16?view=webview2-1.0.4181-prerelease&preserve-view=true#creatediagnosticmonitor)

* [COREWEBVIEW2_DIAGNOSTIC_CATEGORY enum](/microsoft-edge/webview2/reference/win32/webview2experimental-idl?view=webview2-1.0.4181-prerelease&preserve-view=true#corewebview2_diagnostic_category)
   * `COREWEBVIEW2_DIAGNOSTIC_CATEGORY_NETWORK_REQUEST`

* [COREWEBVIEW2_DIAGNOSTIC_SCOPE enum](/microsoft-edge/webview2/reference/win32/webview2experimental-idl?view=webview2-1.0.4181-prerelease&preserve-view=true#corewebview2_diagnostic_scope)
   * `COREWEBVIEW2_DIAGNOSTIC_SCOPE_ENVIRONMENT`
   * `COREWEBVIEW2_DIAGNOSTIC_SCOPE_PROFILE`
   * `COREWEBVIEW2_DIAGNOSTIC_SCOPE_WEBVIEW`

---

<!-- ------------------------------ -->
#### Promotions to Phase 2 (Stable in Prerelease)

No APIs have been promoted from Phase 1: Experimental in Prerelease, to Phase 2: Stable in Prerelease, in this Prerelease SDK.

<!-- ------------------------------ -->
#### Bug fixes

This Prerelease SDK includes the following bug fixes.

<!-- ---------- -->
###### Runtime-only

> * Fixed the `Caption controls` background color setting API.  After this change, to close your app, the app will also need to intercept the `Close` call and handle the call itself.

Author: The `Caption controls` fix was already reported in May Prerelease & Mar. Release.
Find "caption" in rendered page:

Aug Prerelease:
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3855&tabs=dotnetcsharp#runtime-only

May Prerelease:
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3855&tabs=dotnetcsharp#runtime-only-6

March Release (short wording):
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3855&tabs=dotnetcsharp#runtime-only-10

@ojasvi1450

Reviewer: ```suggestion
* Fixed invisible caption button glyphs on dark Window Controls Overlay backgrounds.
```

* Restricted access to a singleton host pipe in legacy WebView2 clients.
* Removed implicit `window.gc` addition on WebViews.
* Fixed the "download shortcut" announcement by screen readers.
* Fixed the Min/Max/Close flicker issue in Windows Control Overlay.
* Fixed honoring a custom scale factor for PDF printing.
* Fixed the anchoring logic for the minimal window download popup.

> 

Reviewer: ```suggestion
* Fixed the inability to drag windows from the top border of custom title bars.
```

   * [ICoreWebView2ExperimentalEnvironment16::CreateDiagnosticMonitor](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalenvironment16?view=webview2-1.0.4181-prerelease&preserve-view=true#creatediagnosticmonitor)

* [COREWEBVIEW2_DIAGNOSTIC_CATEGORY enum](/microsoft-edge/webview2/reference/win32/webview2experimental-idl?view=webview2-1.0.4181-prerelease&preserve-view=true#corewebview2_diagnostic_category)
   * `COREWEBVIEW2_DIAGNOSTIC_CATEGORY_NETWORK_REQUEST`

* [COREWEBVIEW2_DIAGNOSTIC_SCOPE enum](/microsoft-edge/webview2/reference/win32/webview2experimental-idl?view=webview2-1.0.4181-prerelease&preserve-view=true#corewebview2_diagnostic_scope)
   * `COREWEBVIEW2_DIAGNOSTIC_SCOPE_ENVIRONMENT`
   * `COREWEBVIEW2_DIAGNOSTIC_SCOPE_PROFILE`
   * `COREWEBVIEW2_DIAGNOSTIC_SCOPE_WEBVIEW`

---

<!-- ------------------------------ -->
#### Promotions to Phase 2 (Stable in Prerelease)

No APIs have been promoted from Phase 1: Experimental in Prerelease, to Phase 2: Stable in Prerelease, in this Prerelease SDK.

<!-- ------------------------------ -->
#### Bug fixes

This Prerelease SDK includes the following bug fixes.

<!-- ---------- -->
###### Runtime-only

> * Rewrote the test case for `PrintSandboxIframeCompareWithExpected`.

Reviewer: ```suggestion

```

Author: yes, doubted that item, internal

> * Fixed the `Caption controls` background color setting API.  After this change, to close your app, the app will also need to intercept the `Close` call and handle the call itself.

Author: ```suggestion
* Fixed the `Caption control` background color setting API.  After this change, to close your app, the app will also need to intercept the `Close` call and handle the call itself.
```
tech review this suggested change @ojasvi1450

Reviewer: Not required anymore

Author: updated
```
* Fixed invisible caption button glyphs on dark Window Controls Overlay backgrounds.
```

> * Fixed the `Caption` controls background color setting API.  After this change, to close your app, the app will also need to intercept the `Close` call and handle the call itself.

Reviewer: ```suggestion
* Fixed the `Caption controls` background color setting API.  After this change, to close your app, the app will also need to intercept the `Close` call and handle the call itself.
```

Author: also fixed same dup item in section
**Prerelease SDK 1.0.4015-prerelease, for Runtime 149 (May 11, 2026)**
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3855&tabs=dotnetcsharp#runtime-only-6

why dup'd?  re-fixed?  @ojasvi1450 

both sections are Prerelease

Author: Outdated.  [fresh](https://github.com/MicrosoftDocs/edge-developer/pull/3855/files#r3683080893)

* Removed implicit `window.gc` addition on WebViews.
* Fixed the "download shortcut" announcement by screen readers.
* Fixed the Min/Max/Close flicker issue in Windows Control Overlay.
* Fixed honoring a custom scale factor for PDF printing.
* Fixed the anchoring logic for the minimal window download popup.
* Fixed the inability to drag a window via the top border of a custom title bar.

###### SDK-only

* Fixed a WPF crash when changing display topology in composition-controlled apps.

> * Fixed a WebView2/CEF window class collision by using module-unique suffixes.

Reviewer: ```suggestion

```

<!-- end of Prerelease SDK 1.0.4181-prerelease, for Runtime 152 (Aug. 3, 2026) -->

<!-- ====================================================================== -->
## Release SDK 1.0.4129.50, for Runtime 151 (Aug. 3, 2026)

Release Date: Aug. 3, 2026

[NuGet package for WebView2 SDK 1.0.4129.50](https://www.nuget.org/packages/Microsoft.Web.WebView2/1.0.4129.50)

For full API compatibility, this Release version of the WebView2 SDK requires WebView2 Runtime version 151.0.4129.50 or higher.

<!-- ------------------------------ -->
#### Promotions to Phase 3 (Stable in Release)

No additional APIs have been promoted from Phase 2: Stable in Prerelease, to Phase 3: Stable in Release, in this Release SDK.

<!-- ------------------------------ -->
#### Bug fixes

This Release SDK includes the following bug fixes.

<!-- ---------- -->
###### Runtime-only

> * Restricted access to a singleton host pipe in deprecated WebView2.  todo: clarify latter

Reviewer: ```suggestion
* Restricted access to a singleton host pipe in legacy WebView2 clients.
```

This Prerelease SDK includes the following bug fixes.

<!-- ---------- -->
###### Runtime-only

* Rewrote the test case for `PrintSandboxIframeCompareWithExpected`.
* Fixed the `Caption` controls background color setting API.  After this change, to close your app, the app will also need to intercept the `Close` call and handle the call itself.

> * Restricted access to a singleton host pipe in deprecated WebView2.  todo: clarify latter

Author: clarify "in deprecated webview2"

https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3855&tabs=dotnetcsharp#runtime-only

Reviewer: ```suggestion
* Restricted access to a singleton host pipe in legacy WebView2 clients.
```

Author: fixed 2x

###### Runtime-only

* Fixed a bug where disconnecting a screen didn't change the screen resolution correctly.
* Fixed per-monitor DPI in `window.getScreenDetails()`.  ([Issue #4826](https://github.com/MicrosoftEdge/WebView2Feedback/issues/4826))
* Disabled the domain actions component for WebView2.
* Fixed Print-to-PDF API failure when printing PDFs.  ([Issue #5499](https://github.com/MicrosoftEdge/WebView2Feedback/issues/5499))
* Fixed an issue causing Narrator to announce the structural `HWND`, which doesn't have any UI.
* Fixed WebView2 transparency.

> * Fixed invisible caption button glyphs on dark Window Controls Overlay backgrounds.

Reviewer: Oh wait a second - did not realize this is from an older Runtime fix. Can we revert to the original fix?

* Fixed WebView2 transparency.

> * Fixed the API for setting the background color of the **Caption** control.

Author: Short version of same `Caption control` fix, in March Release.  
Use the Aug & May verbose wording (shown in hidden comment below)?
Find "caption" in rendered page:

Aug Prerelease:
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3855&tabs=dotnetcsharp#runtime-only

May Prerelease:
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3855&tabs=dotnetcsharp#runtime-only-6

March Release (short wording):
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3855&tabs=dotnetcsharp#runtime-only-10

Reviewer: ```suggestion
* Fixed invisible caption button glyphs on dark Window Controls Overlay backgrounds.
```
